GTScript-Parallel-model.md

```diff
@@ -0,0 +1,110 @@
+
+To summarize the model, we can say that:
+
+- *computations* are executed sequentially in the order they appear in the code,
+- vertical *intervals* are executed sequentially in the order defined by the *iteration policy* of the  *computation* (where `PARALLEL` means that order is not relevant and thus it will be implementation-specific),
+- *statements* inside *intervals* are executed as (sequential) for-loops 
+  over the `K`-range following the order defined by the iteration policy,
+- a *statement* inside the *interval* is executed as a parallel for-loop over the horizontal dimension(s) with no guarantee on how statements are executed.
+
+#### Example
+On an applied example, this means (by definition `start <= end`):
+
+```python
+with computation(FORWARD):  # Forward computation
+    with interval(start, end):
+        a = tmp[1, 1, 0]
+        b = 2 * a[1, 1, 0]
+
+with computation(BACKWARD):  # Backward computation
+    with interval(start, -2):  # interval A
+        a = tmp[1, 1, 0]
+        b = 2 * a[0, 0, 0]     
+    with interval(-2, end):    # interval B
+        a = 1.1
+        b = 2.2
+        
+with computation(PARALLEL):  # Parallel computation
+    with interval(start, end):
+        a = tmp[1, 1, 0]
+        b = 2 * a[0, 0, 0]
+```
+
+corresponds to the following pseudo-code:
+
+```python
+# Forward computation
+for k in range(start, end):
+    parfor ij:
+        a[i, j, k] = tmp[i+1, j+1, k]
+    parfor ij:
+        b[i, j, k] = 2 * a[i+1, i+1, k]
+
+# Backward computation
+for k in reversed(range(end-2, end)):  # interval B
+    parfor ij:
+        a[i, j, k] = 1.1
+    parfor ij:
+        b[i, j, k] = 2.2
+for k in reversed(range(start, end-2)):  # interval A
+    parfor ij:
+        a[i, j, k] = tmp[i+1, j+1, k]
+    parfor ij:
+        b[i, j, k] = 2 * a[i, j, k]
+
+# Parallel computation
+parfor k in range(start, end):
+    parfor ij:
+        a[i, j, k] = tmp[i+1, j+1, k]
+    parfor ij:
+        b[i, j, k] = 2 * a[i, j, k]
+```
+
+where `parfor` means that there is no guarantee of the order in which the iteration is performed. Additionally, the following restrictions apply:
+
+- No self-assignments if `i != 0` or `j != 0`
```

Comment:
Can be expanded into 
* what self-assignment means (namely, reading and writing withtin the same interval, not just the same statement)
* `k!=0` is also not allowed if PARALLEL
* and that the `i!=0` is understood in the read offsets

Reply:
I think actually the whole sentence 

> Additionally, the following restrictions apply: ...

should be removed as it doesn't add extra information. Or if you like to have this information somewhere, I'd rephrase this as a consequence from the parallel model.

Reply:
done